ai_wars/game_class.py

```diff
@@ -45,51 +63,44 @@ def main_loop(self) -> None:
 			self.delta_time = self.clock.tick(144)
 			self.time_elapsed_since_last_action += self.delta_time
 
+			self._handle_inputs()
 			self._handle_events()
 			self._process_game_logic()
 			self._draw()
 
-	def _handle_events(self) -> None:
-		"""Private helper method to listen for and process input via
-			pygame events
-		"""
-		for event in pygame.event.get():
-			match event:
-				# check if the game should be closed
-				case event if event.type == pygame.QUIT or \
-					 (event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE):
-					sys.exit()
-
-				# check if the space bar got pressed -> shoot
-				case _ if event.type == pygame.KEYDOWN and event.key == pygame.K_SPACE:
-					if self.spaceship:
-						self.spaceship.shoot()
-
-				# decrease the score of the players (event gets fired every second)
-				case _ if event.type == pygame.USEREVENT+0:
-					for spaceship in self.spaceships:
-						spaceship.score -= 1
-
+	def _handle_inputs(self) -> None:
+		"""private method to process inputs and limit the bullet frequency"""
 		# check whick keys eare pressed
 		is_key_pressed = pygame.key.get_pressed()
 
 		match is_key_pressed:
+			case is_key_pressed if is_key_pressed[pygame.K_SPACE]:
+				# limit the frequency of bullets
+				if self.time_elapsed_since_last_action > self.SHOOT_COOLDOWN:
+					self.spaceship1.action(EnumAction.SHOOT)
+					self.time_elapsed_since_last_action = 0
 			case is_key_pressed if is_key_pressed[pygame.K_LEFT]:
-				self.spaceship.move(EnumAction.LEFT)
+				self.spaceship1.action(EnumAction.LEFT)
 			case is_key_pressed if is_key_pressed[pygame.K_RIGHT]:
-				self.spaceship.move(EnumAction.RIGHT)
+				self.spaceship1.action(EnumAction.RIGHT)
 			case is_key_pressed if is_key_pressed[pygame.K_UP]:
-				self.spaceship.move(EnumAction.FORWARD)
+				self.spaceship1.action(EnumAction.FORWARD)
 			case is_key_pressed if is_key_pressed[pygame.K_DOWN]:
-				self.spaceship.move(EnumAction.BACKWARDS)
+				self.spaceship1.action(EnumAction.BACKWARD)
 
-	def _draw_leaderboard(self, screen) -> None:
-		"""private method to draw the leaderboard on the given screen"""
-		for pos, (player, score) in enumerate(self.leaderboard.items()):
-			score_string = f"{player} : {score}"
-			text_surface = self.font.render(score_string, False, (255, 255, 255))
-			screen.blit(text_surface, (0, self.font_height*pos))
+	def _handle_events(self) -> None:
+		"""Private helper method to listen and process pygame events"""
+		for event in pygame.event.get():
+			match event:
+				# check if the game should be closed
+				case event if event.type == pygame.QUIT or \
+					 (event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE):
+					sys.exit()
 
+				# decrease the score of the players (event gets fired every second)
+				case _ if event.type == pygame.USEREVENT+0:
```

Comment:
```suggestion
				case _ if event.type == DECREASE_SCORE_EVENT:
```
Ist vielleicht so etwas schöner zu lesen.